glean-core/src/ping/mod.rs

```diff
@@ -124,7 +124,7 @@ impl PingMaker {
             let mut file = File::create(&temp_ping_path)?;
             file.write_all(url_path.as_bytes())?;
             file.write_all(b"\n")?;
-            file.write_all(ping_content.as_bytes())?;
+            file.write_all(::serde_json::to_string(ping_content)?.as_bytes())?;
```

Comment:
Wait, that works with `std::io::Result` here? I'm confused, I would have though we need to switch to our own error type here, but if compile & tests pass it's ok (I'll take a look later to see what's going on)

Reply:
Oh, yeah, I'm confused, too, come to think of it.

Reply:
This is why: https://docs.rs/serde_json/1.0.39/src/serde_json/error.rs.html#134